### DISCUSSION TITLE
Changing SW I2C pins at runtime

### DISCUSSION THREAD

**Discussion by strud** [olikraus/u8g2#2495] [Q&A]
I am using U8g2 with a ST7567 in SW I2C mode

Constructor:

U8G2_ST7567_JLX12864_F_SW_I2C u8g2(U8G2_R2, SCL2, SDA2);

My question: is there a way without modifying the library to change the clock and data lines to other pins at run time?

I have a configurable system and 2x I2C buses so I would like the option of having the GUI put onto either of these buses.

**Comment by olikraus** [comment]
Yes, cou can assign new Arduino pin numbers:
``` C++
u8g2.GetU8x8()->pins[U8X8_PIN_I2C_CLOCK] = <i2c  clock arduino pin number>
u8g2.GetU8x8()->pins[U8X8_PIN_I2C_DATA] = <i2c data arduino pin number>
```

Reference: 
https://github.com/olikraus/u8g2/blob/bdb7896cb1c1966ace20b646999d48987cdd57c3/csrc/u8x8.h#L320-L321

https://github.com/olikraus/u8g2/blob/bdb7896cb1c1966ace20b646999d48987cdd57c3/csrc/u8x8.h#L370
